pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -25,18 +25,24 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
+	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 	"k8s.io/kubernetes/pkg/scheduler/apis/config/validation"
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/helper"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/names"
+	"k8s.io/kubernetes/pkg/scheduler/util"
 )
 
 // NodeAffinity is a plugin that checks if a pod node selector matches the node label.
 type NodeAffinity struct {
 	handle              framework.Handle
 	addedNodeSelector   *nodeaffinity.NodeSelector
 	addedPrefSchedTerms *nodeaffinity.PreferredSchedulingTerms
+
+	// logger is only meant to be used by background activities which don't
+	// have some other logger in their parent callstack.
+	logger klog.Logger
```

Comment:
should we change the `QueueingHintFn` to pass a context?

Reply:
I would prefer to pass a logger, for performance reasons.

Reply:
Sounds good, passing a logger makes sense too.

Reply:
@alculquicondor Can I help do that?

Reply:
See https://github.com/kubernetes/kubernetes/pull/119242

Reply:
To pass a logger sounds good. `/lgtm`-ed @carlory 's PR.

Reply:
> I would prefer to pass a logger, for performance reasons.

I remember we talked about `ctx` somewhere in the past but couldn't find the thread.
Why do we eliminate `ctx` from the parameters? In wasm project, we're working on QueueHint support and communication to the guest needs ctx. Of course, we can create `context.Background()` in our QueueHint though, it's at least one usecase of ctx in QueueHint parameters.